adapters/brightroll/brightroll.go

```diff
@@ -0,0 +1,170 @@
+package brightroll
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strconv"
+)
+
+type BrightrollAdapter struct {
+	URI string
+}
+
+func (a *BrightrollAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		err := &adapters.BadInputError{
+			Message: "No impression in the bid request",
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	validImpExists := false
+
+	for _, imp := range request.Imp {
+		//Brightroll supports only banner and video impressions as of now
+		if imp.Banner != nil {
+			validImpExists = true
+		} else if imp.Video != nil {
+			validImpExists = true
+		} else {
+			err := &adapters.BadInputError{
+				Message: fmt.Sprintf("Brightroll only supports banner and video imps. Ignoring imp id=%s", imp.ID),
+			}
+			errs = append(errs, err)
+		}
+	}
+
+	if !validImpExists {
+		err := &adapters.BadInputError{
+			Message: fmt.Sprintf("No valid impression in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+	errors := make([]error, 0, 1)
+
+	var bidderExt adapters.ExtImpBidder
+	err = json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+
+	if err != nil {
+		err = &adapters.BadInputError{
+			Message: "ext.bidder not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	var brightrollExt openrtb_ext.ExtImpBrightroll
+	err = json.Unmarshal(bidderExt.Bidder, &brightrollExt)
+	if err != nil {
+		err = &adapters.BadInputError{
+			Message: "ext.bidder.publisherName not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if brightrollExt.PublisherName == "" {
+		err = &adapters.BadInputError{
+			Message: "publisherName is empty",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	thisURI := a.URI
+	thisURI = thisURI + "?publisher=" + brightrollExt.PublisherName
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(request.Device.DNT)))
+	}
+	fmt.Println("**** URI ::: " + thisURI)
```

Comment:
Looks like this slipped into your most recent commits? It shouldn't be here though... it'll pollute the app logs.